### PR TITLE
Cherry-pick new UI

### PR DESCRIPTION
**Pull request by vpavlin**
I cherry picked all the commits related to the new UI

I also tweaked the UI config so that the UI works properly without having the `ui` section in the config

### CODE REVIEW

File: jupyterhub_singleuser_profiles/ui/package.json [hunk 1]
```diff
@@ -1,39 +1,101 @@
 {
-  "name": "jupyterhub-singleuser-profiles-ui",
+  "name": "jupyterhub_singleuser_profiles_ui",
   "version": "0.1.0",
+  "description": "JupyterHub Spawner UI",
+  "author": "Red Hat",
```
Comment (jeff-phillips-18) [inline jupyterhub_singleuser_profiles/ui/package.json:5]: Should this be Red Hat upstream?

Comment (vpavlin) [reply jupyterhub_singleuser_profiles/ui/package.json:5]: We could also put `Open Data Hub` there, but I think Red Hat is fine

**Review by jeff-phillips-18 — APPROVED** [review]
LGTM

**Review by mroman-redhat — APPROVED** [review]
lgtm!

**Pull request merged**